modules/db/models/community.js

```diff
@@ -0,0 +1,163 @@
+    managerWalletAddress: {type: String},
+    usersWalletAddress: {type: String},
+    mnemonic: {type: String},
+    defaultCurrency: {type: Schema.Types.ObjectId, ref: 'Currency'},
+    managerBalance: {type: BalanceSchema},
+    usersBalance: {type: BalanceSchema}
+  }).plugin(timestamps())
+
+  BalanceSchema.set('toJSON', {
+    getters: true,
+    virtuals: true,
+    transform: (doc, ret, options) => {
+      const safeRet = {
+        id: ret._id.toString(),
+        createdAt: ret.created_at,
+        updatedAt: ret.updated_at,
+        currency: ret.currency,
+        blockchainBalance: ret.blockchainBalance,
+        offchainBalance: ret.offchainBalance
+      }
+      return safeRet
+    }
+  })
+
+  CommunitySchema.set('toJSON', {
+    getters: true,
+    virtuals: true,
+    transform: (doc, ret, options) => {
+      const safeRet = {
+        id: ret._id.toString(),
+        createdAt: ret.created_at,
+        updatedAt: ret.updated_at,
+        managerWalletAddress: ret.managerWalletAddress,
+        usersWalletAddress: ret.usersWalletAddress,
+        // mnemonic: ret.mnemonic,
+        defaultCurrency: ret.defaultCurrency,
+        managerBalance: ret.managerBalance,
+        usersBalance: ret.usersBalance
+      }
+      return safeRet
+    }
+  })
+
+  const Community = db.model('community', CommunitySchema)
+
+  function community () {}
+
+  community.create = (data) => {
+    return new Promise((resolve, reject) => {
+      const community = new Community(data)
+      community.save((err, newObj) => {
+        if (err) {
+          return reject(err)
+        }
+        if (!newObj) {
+          let err = 'Community not saved'
+          return reject(err)
+        }
+        resolve(newObj)
+      })
+    })
+  }
+
+  community.getById = (id) => {
+    return new Promise((resolve, reject) => {
+      Community.findById(id, (err, doc) => {
+        if (err) {
+          return reject(err)
+        }
+        if (!doc) {
+          err = `Community with not found for id ${id}`
+          return reject(err)
+        }
+        resolve(doc)
+      })
+    })
+  }
+
+  community.getByManagerWalletAddress = (address) => {
+    return new Promise((resolve, reject) => {
+      Community.findOne({managerWalletAddress: address}, (err, doc) => {
+        if (err) {
+          return reject(err)
+        }
+        if (!doc) {
+          err = `Community not found for managerWalletAddress: ${address}`
+          return reject(err)
+        }
+        resolve(doc)
+      })
+    })
+  }
+
+  community.getByUsersWalletAddress = (address) => {
+    return new Promise((resolve, reject) => {
+      Community.findOne({usersWalletAddress: address}, (err, doc) => {
+        if (err) {
+          return reject(err)
+        }
+        if (!doc) {
+          err = `Community not found for usersWalletAddress: ${address}`
+          return reject(err)
+        }
+        resolve(doc)
+      })
+    })
+  }
+
+  community.updateBalance = () => {
```

Comment:
We need to add another scheme called internalTransaction that manage the stages to transferring balance to retain atomicity.
It will have states and will make sure both sides of the transfer will update their balance, or non of them will.

Reply:
This will be a part of the transaction management schemas (separate PR)